### ISSUE TITLE
A manual resolve redirects button

### ISSUE THREAD

**Issue by ShrirajHegde** [LinkSheet/LinkSheet#401]
### Feature description

A manual redirection resolution is very useful for links from YouTube app and Twitter app for sharing purposes.

This is a great alternative to always resolving redirection since it is not very responsive and hence a worse UX.

**Comment by 1fexd** [comment]
What's the issue with "automatically" following redirects?

**Comment by ShrirajHegde** [comment]
It's usually slow. The loading screen takes 500ms - 1s depending on connection latency and feels like a road bump when being used to quick popups without any link operations.

Another option is to asynchronously follow redirects and swap them on resolution instead of the loading screen. This also adds the snappiness back, and users can visually process the app choices available while the link is resolved.

**Comment by 1fexd** [comment]
True, yeah. I don't think doing it asynchronously is a good idea since apps usually aren't able to handle links which do an auto redirect (e.g. url shorteners), but adding a button to only follow redirects when the user wishes to do so is a good idea.

**Comment by PhilC813** [comment]
Would also hope to see this manual approach at some point!

**Comment by 1fexd** [comment]
I've added a button called "Follow redirects" to the bottom sheet - to have it show up, you'll need to enable (if not yet enabled) the existing experiments `Improved intent resolver` and `Loop detector`, as well as the new `Manual redirect resolving` experiment. Make sure to toggle off "Settings" -> "Links" -> " Follow redirects" since otherwise redirects will be resolved automatically before the bottom sheet shows up

**Comment by PhilC813** [comment]
Thank you so much for adding this! It seems to work as expected. One improvement would be to hide the button from the next sheet after it's been used.

**Comment by 1fexd** [comment]
> Thank you so much for adding this! It seems to work as expected. One improvement would be to hide the button from the next sheet after it's been used.

I'll consider that in the future, thanks - closing this for now to clean up the issue tracker
